lib/kerror/codes/2-api.json

```diff
@@ -140,6 +140,12 @@
           "code": 9,
           "message": "Admin user is already set.",
           "class": "PreconditionError"
+        },
+        "incomplete_multiple_request": {
+          "description": "Failed to execute some or all actions requested",
+          "code": 10,
+          "message": "At least one of the %s actions failed.",
+          "class": "PartialError"
```

Comment:
You shouldn't throw a PartialError, because that error type has a 206 status code. This, in fact, should be a Partial Success return type, and not an error class (the PartialError class is a bit of an anomaly, we might remove it in the future)

Reply:
Oh this makes sense but it questions the purpose of the issue, @Aschen may had another thing in mind while creating this issue... Should we discuss about this?

Reply:
It should be a BadRequestError to me, even if some documents succeed to be written

Reply:
Hm, bad requests are more for erroneous requests (invalid syntax/payload).

I don't have a ready-to-use solution, so here are a few thoughts:
* we do have individual error, for each failed document. If there is 1 error, or if all errors are of the same type, we could throw that error instead. Problem is, our current error system does not handle polymorphic errors
* we could have some kind of error wrapper, used to aggregate multiple errors. That could be useful for the validation module (when we rework it). Let's call it `MultipleError` for the sake of example (name to be debated). Since this is not something that exists in HTTP error codes, and since we don't know what kind of underlying error it contains, we could use the generic `400` error code. I know it's the BadRequest one, but I lack better idea (least we use a currently unused HTTP code for our own needs)

Reply:
I like the second option :+1:

Reply:
Let's go for it then, I'll opt for 400 for this MultipleErrorsError

Reply:
`MultipleError` ?